src/ufuncs.c

```diff
@@ -235,31 +258,52 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
                      npy_intp *steps, void *data)
 {
     FuncGEOS_Y_Y *func = (FuncGEOS_Y_Y *)data;
-    GEOSGeometry *in1, *ret_ptr;
+    GEOSGeometry *in1;
```

Comment:
Agree on the `geom_arr` declaration, I am not sure I follow on the NULL initialization. Both `in1` and `geom_arr` are assigned to below. What is the use of the `NULL`? Just good practice?

Reply:
I think in general it is a good practice to assign pointers to `NULL` esp. where they are used in a conditional where it is possible they might not have been assigned.  I think the GCC compiler emits warnings about that.

Since there is `in1 == NULL` below, even though it follows assignment by way of `get_geom(*(GeometryObject **)ip1, &in1)`, there is always the chance that it isn't assigned by that function to `NULL` or a valid value by that function.  Of course, in this specific case, if that function wasn't setting it to `NULL` there would be bigger problems (since it is in a loop where some items may be `NULL`).

Reply:
We shouldn't rely on a function to initialize the variable, that is true. I added `= NULL` at a couple of places in the code